src/component/SelectSearch.test.tsx

```diff
@@ -1,39 +1,111 @@
+import axiosMock from 'axios';
 import * as React from 'react';
-import { mount } from 'enzyme';
-import SelectSearch, { SelectSearchProps } from './SelectSearch';
-import { SearchResult } from '../services/ComponentMap';
-import { GROUP_PLACEHOLDER, GROUP_NOT_FOUND } from './form/constants';
-
-const { endpoints } = require('../../assets/config');
-
-describe('SelectSearch >', () => {
-    describe('render >', () => {
-        describe('split by group membership >', () => {
-            const props: SelectSearchProps = {
-                url: endpoints.groups,
-                name: 'Group',
-                resultType: 'group',
-                placeholder: GROUP_PLACEHOLDER,
-                searchPromptText: GROUP_NOT_FOUND,
-                closeOnSelect: false,
-                initial: []
-            };
-
-            const loadOptionsSpy = jest.spyOn(SelectSearch.prototype, 'loadOptions' as any);
-
-            const wrapper = mount(<SelectSearch {...props} />);
-
-            it('should render <Async />, pass it expected props', () => {
-                expect(wrapper.find('Async').props()).toEqual(
-                    expect.objectContaining({
-                        name: props.name,
-                        placeholder: props.placeholder,
-                        searchPromptText: props.searchPromptText,
-                        closeOnSelect: props.closeOnSelect
-                    })
+import { FlowEditorConfig } from '../flowTypes';
+import { composeSetup, Resp, restoreSpies, yieldToPromises } from '../testUtils';
+import { resultsToSearchOptions } from '../utils';
+import { GROUP_NOT_FOUND, GROUP_PLACEHOLDER } from './form/constants';
+import SelectSearch, { LOADING_PLACEHOLDER, SelectSearchProps } from './SelectSearch';
+
+const { endpoints } = require('../../assets/config') as FlowEditorConfig;
+const groupsResp = require('../../assets/groups.json') as Resp;
+
+const baseProps = {
+    url: endpoints.groups,
+    name: 'Groups',
+    resultType: 'groups',
+    placeholder: GROUP_PLACEHOLDER,
+    searchPromptText: GROUP_NOT_FOUND
+};
+
+const setup = composeSetup<SelectSearchProps>(baseProps, null, SelectSearch);
+
+const mapRespToSearchOptions = (resp: Resp) => resp.results.map(resultsToSearchOptions);
+
+const COMPONENT_TO_TEST = SelectSearch.name;
+
+describe(`${COMPONENT_TO_TEST}`, () => {
+    describe('render', () => {
+        it('should render self, children with required props', async () => {
+            const selectRefSpy = jest.spyOn(SelectSearch.prototype, 'selectRef');
+            const loadOptionsSpy = jest.spyOn(SelectSearch.prototype, 'loadOptions');
+            const searchSpy = jest.spyOn(SelectSearch.prototype, 'search');
+            const { wrapper, props: { name, placeholder, searchPromptText, url } } = setup();
+            const SelectSearchInstance = wrapper.instance();
+            const asyncSelect = wrapper.find('Async');
+
+            await yieldToPromises();
```

Comment:
yielding here because `SelectSearch.search` is called when `axios.get` in `SelectSearch.loadOptions` resolves

Reply:
eh, if we all get the picture i figure the comment above `yieldToPromises` is informative enough. wasn't sure if this was a new or confusing pattern but i get the sense it isn't

Reply:
That there’s a yield when it’s not obvious there should be a request is the comment worthy bit.

Reply:
ah gotcha